### DISCUSSION TITLE
Items api response - respect sorting of backend

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#7243] [Archive]
If I make an api request to retrive all content of an items, response field not respect sorting in Directus web app.

I think that the respect of sorting is mandatory if someone would like to use Directus as a page builder for front end. Do you know if this behavior could be set?

I attach a screenshot for better explain the problem.

<img width="1614" alt="Schermata 2021-08-06 alle 12 00 54" src="https://user-images.githubusercontent.com/30528153/128494719-1e68de09-bfc6-41d7-a727-8397ba82ed34.png">

**Comment by benhaynes** [comment]
This is an interesting question. The Field Sort is really meant for App Form Layout... but whether or not that should also apply to the API response's JSON is up to @rijkvanzanten 

For Page Builders, you would use a page builder field/interface, which does honor/respect the value order and hierarchy.

**Comment by nickrum** [comment]
Javascript/JSON objects are unordered by design. It is not recommended to rely on their ordering.

The better approach would be to fetch the field ordering separately and sort the data based on that.
